apps/prairielearn/src/pages/studentAssessmentInstance/studentAssessmentInstance.js

```diff
@@ -244,7 +246,6 @@ router.get(
     );
     res.locals.assessment_text_templated = assessment_text_templated;
 
-    res.locals.showTimeLimitExpiredModal = req.query.timeLimitExpired === 'true';
```

Comment:
Cleanup: it turns out this modal never actually got a chance to render because `middlewares/studentAssessmentAccess.ts` would render a response before we ever got to this page. I also removed the corresponding modal from the template.

Do we think this modal adds value? If so, we can update the middleware to render it. If not, I'll do an extra bit of cleanup and remove the query parameter from here:

https://github.com/PrairieLearn/PrairieLearn/blob/bf96affaace6514e83c62cd828ad2083749fea06/apps/prairielearn/src/pages/studentAssessmentInstance/studentAssessmentInstance.js#L181

Reply:
In the past I noticed that this didn't show for PrairieTest time limits, but it did for access rules time limits. I'm not sure if this is still the case since I'm using PrairieTest only. Anyway, it may depend on if "show closed assessment" is enabled.

Reply:
I tested this locally by setting a time limit to 1 minute, starting the assessment, and then waiting. The modal did not show, and indeed the middleware is what rendered the response. Do you know of another way to test this?

Reply:
I just tested it locally with an assessment where `showClosedAssessment` was true in the appropriate access rule, and it showed the modal.

Reply:
Good find! I'll repeat this question:

> Do we think this modal adds value? If so, we can update the middleware to render it.

IMO we should either always show it when the time limit expires, or never show it.

Reply:
I'd be fine with an update to the middleware, as long as we also ensure that whatever error message is shown when access is removed via PrairieTest expiry is also accounted for (is it the same middleware? I haven't checked).

A flash message might work too, and may be more robust.

Reply:
I thought about a flash message and it's actually worse: if a student has the assessment instance and three questions open when the time limit runs out, we'll end up with 4 roughly-simultaneous page submits + loads. With flashes, we could end up with 4 flashes on one page and none on the others, and that'll be essentially random.

I talked with @mwest1066 and he wants to keep the modals. I'll add it back and update the middleware page to render it too.

Reply:
I added the modal back here: ca93aef